### PR TITLE
Dynamic AllowedConfiguration

### PR DESCRIPTION
**Pull request by yashvardhan-n**

### CODE REVIEW

File: src/types/ts-to-chart-event.types.ts [hunk 1]
```diff
@@ -191,6 +192,8 @@ export interface InitializeEventResponsePayload {
      * @version SDK: 0.1 | ThoughtSpot:
      */
     visualPropEditorDefinition?: VisualPropEditorDefinition;
+    // Whether user wants thoughtspot default number and conditional formatting
```
Comment (chetan1507) [inline src/types/ts-to-chart-event.types.ts:195]: lets add a more generic message here. 
Toggle native configurations supported by TS UI. Ex: column level number and conditional formatting.

Also add versioning details like below.
/**
     * visual properties editor definition
     *
     * @version SDK: 0.1 | ThoughtSpot:
     */

**Review by chetan1507 — CHANGES_REQUESTED** [review]
unit tests.

File: src/main/custom-chart-context.ts [hunk 2]
```diff
@@ -127,6 +135,10 @@ const DEFAULT_CHART_CONTEXT_PROPS: Partial<CustomChartContextProps> = {
     validateConfig: () => ({ isValid: true }),
     validateVisualProps: () => ({ isValid: true }),
     chartConfigEditorDefinition: undefined,
+    allowedConfigurations: {
+        allowNumberFormatting: false,
```
[the remark this answers is not in the document]
Comment (yashvardhan-n) [reply src/main/custom-chart-context.ts]: ack

File: src/types/ts-to-chart-event.types.ts (hunk 1 above)
Comment (yashvardhan-n) [reply src/types/ts-to-chart-event.types.ts]: ack

**Review by Rohit1508 — APPROVED** [review]

**Pull request merged**